### ISSUE TITLE
Able to set client cookie from server generated cookie?

### ISSUE THREAD

**Issue by Ajaay** [VeliovGroup/Meteor-Cookies#17]
Hi,

I am generating cookies with AWS on the Accounts.Login hook on the server, and want to be able to then set these cookies on the client.

Is this possible with this package?

I've followed both documented example setups, but I'm not able to retrieve them on the client after they've been set on the server.

I see there is a 'send' method to forward cookies from client to server, but I really need to reverse of this.

Thanks!

**Comment by dr-dimitru** [comment]
@Ajaay you can use `.send()` method to synchronize server and client cookies.
Use `opts.onCookies` hook for this purpose, __note__: *cookies on the server are stateless*

**Comment by Ajaay** [comment]
According to the docs this is only available to the Client?

Are you saying that if I initiate send from the client and then catch it on the server with the opts.onCookies hook that I can then respond back to the client with a new set of cookie values?

Thanks

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by dr-dimitru** [comment]
@Ajaay yes, just set new cookies on this hook, and it will be delivered to the Client.
Let me know if you will have any issues with it

**Comment by Ajaay** [comment]
Ok I’ll give it a shot. My only worry with that is how I can ensure the client initiates the send after the server has received the cookie data from AWS..

Perhaps I just need to keep retrying on the client until a value is set successfully...

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by dr-dimitru** [comment]
Maybe you should use pub/sub to reactively send data to the user once data received from AWS?

You don't have to use "real" MongoDB collection, see "counts" example in Meteor's docs

**Comment by Ajaay** [comment]
Ok yeh that looks like it could do it - great shout thanks.

Just working through the send - onCookies hook and I've managed to get something set in the client. What I'm seeing though is that if I try to set more than one cookie, only the last set call finds it's way to the client. For example:

```
const cookie = new Cookies({
  auto: true,
  onCookies(cookies) {
    cookies.set('CloudFront-Key-Pair-Id', CloudFrontKeyPairId.value, options);
    cookies.set('CloudFront-Policy', CloudFrontPolicy.value, options);
    cookies.set('CloudFront-Signature', CloudFrontSignature.value, options);
  }
});
```

Only CloudFront-Signature ends up on the client. I've checked it's not an issue with what I'm setting in the others because commenting any of them out, it's always the last one that gets set..

Trying to work my way through the source code to figure out why this might be but struggling at the moment.. 

Any ideas?

Thanks for your help!

**Comment by dr-dimitru** [comment]
@Ajaay thank you for reporting about this, labelled as "maybe" [bug]
Going to test it

**Comment by Ajaay** [comment]
@dr-dimitru My current theory is that it's because we're using this.response.setHeader in the set method, which will only allow us to set a single cookie. Perhaps we can create a valid cookieString for each of this.cookies and then use this.response.writeHead to send them all at once?

**Comment by Ajaay** [comment]
@dr-dimitru Found something interesting while trying to troubleshoot this today... If I browse directly to http://localhost:3000/__cookie__/set , all 3 cookies appear in the browser. 

I'm doing some packet captures on my machine to dig into the HTTP headers but at the moment I can't spot any difference between what is being sent and received from the 'send' method, and then when browsing direct...

It's a strange one.

**Comment by dr-dimitru** [comment]
Hello @Ajaay I'm sorry it's been a while, have you solved this issue? How?

**Comment by dr-dimitru** [comment]
Hello @Ajaay ,

Thank you for submitting this issue! Bug was successfully fixed in [`v2.6.1`](https://github.com/VeliovGroup/Meteor-Cookies/releases/tag/2.6.1).
Please, upgrade and let me know if issue is fixed on your end.
